pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -240,7 +241,7 @@ func (r *Reconciler) runReconcileSeedFlow(
 		deployPersesCRD = g.Add(flow.Task{
 			Name:   "Deploying Perses-related custom resource definitions",
 			Fn:     component.OpWait(c.persesCRD).Deploy,
-			SkipIf: seedIsGarden,
+			SkipIf: seedIsGarden || !features.DefaultFeatureGate.Enabled(features.Perses),
```

Comment:
This feels wrong. In `newPersesOperator(...)` there is explicit handling of the feature gate, i.e. the component either deploys or destroys depending on the feature gate, but if the task is skipped the deploy/destroy is not executed. Hence, `perses-operator` and its CRDs might stay in the seed cluster even if the feature gate is disabled.

Reply:
Good point, the `newPersesOperator(...)` call existed before I realized I can use `SkipIf`.

Reply:
I am not sure if you should use `SkipIf` at all. You want to remove `perses-operator` if the feature gate is disabled. The achieve this there needs to be some action, i.e. either you need to execute the task regardless and implement the condition there or you would need an inverse flow task for destroying the resources.
In other components I am aware of, we simply used the former approach.